### PR TITLE
Add CQL support

### PR DESCRIPTION
**Pull request by SpencerC**
* Add CQL flavor.
* Add support for CQL argument compilation.
* Adopt substests for interpolation tests.
* Add support for CQL interpolation.
* Add support for CQL blobs: https://docs.datastax.com/en/cql-oss/3.x/cql/cql_reference/blob_r.html
* Add CQL timestamp interpolation support: https://docs.datastax.com/en/cql-oss/3.x/cql/cql_reference/timestamp_type_r.html
* Add builder test for CQL.
* Add support for the NOW function: https://docs.datastax.com/en/cql-oss/3.3/cql/cql_reference/timeuuid_functions_r.html?hl=now
* Add support for CQL update IF statement: https://docs.datastax.com/en/cql-oss/3.3/cql/cql_reference/cqlUpdate.html#Conditionallyupdatingcolumns
* Drop table prefixes for CQL SELECTs.
* Switch to single quoting for CQL queries: https://docs.datastax.com/en/cql-oss/3.3/cql/cql_reference/escape_char_r.html
* Update documentation.
* Use valid CQL in tests.
* Support CQL LIMIT.

### CODE REVIEW

File: update.go [hunk 1]
```diff
@@ -157,6 +159,13 @@ func (ub *UpdateBuilder) Limit(limit int) *UpdateBuilder {
 	return ub
 }
 
+// If sets IF expressions for UPDATE.
+func (ub *UpdateBuilder) If(andExpr ...string) *UpdateBuilder {
```
Comment (huandu) [inline update.go:163]: It's not recommended to add this method. `IF` is a CQL-specific extension. AFAIK, there is no other DBMS supporting this. I recommend you to implement this in your CQL business code.

```go
func CQLUpdateIf(ub *sqlbuilder.UpdateBuilder, andExpr ...string) *sqlbuilder.UpdateBuilder {
    ub.SQL("IF")
    ub.SQL(strings.Join(andExprs, " AND "))
    return ub
}
```

As you can see, this package mainly defines methods for keywords available in ISO SQL-2016 spec, which is a subset of all DBMS implementations, with a few exceptions. I do think it's not convenient for users who build SQL with only one flavor. There is a long standing issue #60 (in Chinese) in which the support to `ON DUPLICATE KEY UPDATE` is requested. I cannot resolve this issue as SQL syntax for duplicate key differs a lot in different DBMS. It's hard to define a unified method for this feature.

A possible way to support flavor-specific keywords is to create sub-package for every flavor, e.g. `github.com/huandu/go-sqlbuilder/cql` for CQL, and then define methods based on the flavor's SQL spec. However, there will be a lot of work to do so.

**Review by huandu — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution. I leave some review comments here. We can discuss in this PR thread.

**Comment by huandu** [comment]
@SpencerC Could you update your PR per my review comments? Feel free to comment on my comments.

**Comment by SpencerC** [comment]
Yep, have some other work to finish first.

**Comment by huandu** [comment]
Got it.

**Review by huandu — CHANGES_REQUESTED** [review]
There are some compile errors in test case. Please fix them. Thanks.

**Comment by SpencerC** [comment]
Apologies, should be fixed now.

**Comment by huandu** [comment]
LGTM. Thanks a lot for your time, @SpencerC !

**Review by huandu — COMMENTED** [review]
LGTM.

**Pull request merged**